test/sql/catalog/dependencies/test_write_after_rename.test

```diff
@@ -0,0 +1,25 @@
+# name: test/sql/catalog/dependencies/test_write_after_rename.test
+# group: [dependencies]
+
+load __TEST_DIR__/create_after_rename.db
+
+statement ok
+create table tbl (i integer);
+
+statement ok
+insert into tbl VALUES (5);
+
+statement ok
+begin transaction;
+
+statement ok
+alter table tbl rename to tbl2;
+
+statement ok
+create table tbl (b varchar);
+
+statement ok
+insert into tbl VALUES ('test')
+
+statement ok
+commit;
```

Comment:
Could we add a check with restart here that actually verifies this is written to disk correctly?